src/components/cursor-controller.js

```diff
@@ -0,0 +1,393 @@
+  },
+
+  _handleMouseUp: function() {
+    const lookControls = this.data.camera.components["look-controls"];
+    if (lookControls) lookControls.play();
+    this.data.cursor.emit(this.data.controllerEndEvent, {});
+    this._endTeleport();
+  },
+
+  _handleWheel: function(e) {
+    if (this.isGrabbing) {
+      switch (e.deltaMode) {
+        case e.DOM_DELTA_PIXEL:
+          this.currentDistanceMod += e.deltaY / 500;
+          break;
+        case e.DOM_DELTA_LINE:
+          this.currentDistanceMod += e.deltaY / 10;
+          break;
+        case e.DOM_DELTA_PAGE:
+          this.currentDistanceMod += e.deltaY / 2;
+          break;
+      }
+    }
+  },
+
+  _handleEnterVR: function() {
+    if (AFRAME.utils.device.checkHeadsetConnected()) {
+      this.inVR = true;
+      this._updateController();
+    }
+  },
+
+  _handleExitVR: function() {
+    this.inVR = false;
+    this._updateController();
+  },
+
+  _handleRaycasterIntersection: function(e) {
+    this.data.cursor.emit("raycaster-intersection", e.detail);
+  },
+
+  _handleRaycasterIntersectionCleared: function(e) {
+    this.data.cursor.emit("raycaster-intersection-cleared", e.detail);
+  },
+
+  _handleControllerEvent: function(e) {
+    const isInteractable = this._isTargetOfType(TARGET_TYPE_INTERACTABLE) && !this.grabStarting;
+    if (isInteractable || this._isTargetOfType(TARGET_TYPE_UI)) {
+      this.grabStarting = true;
+      this.data.cursor.emit(this.data.controllerEvent, e.detail);
+    } else if (e.type !== this.data.grabEvent) {
+      this._startTeleport();
+    }
+  },
+
+  _handleControllerEndEvent: function(e) {
+    if (this.isGrabbing || this._isTargetOfType(TARGET_TYPE_UI)) {
+      this.grabStarting = false;
+      this.data.cursor.emit(this.data.controllerEndEvent, e.detail);
+    } else if (e.type !== this.data.releaseEvent) {
+      this._endTeleport();
+    }
+  },
+
+  _handleModelLoaded: function() {
+    this.physicalHand = this.data.playerRig.querySelector(this.data.physicalHand);
+  },
+
+  _handleCursorLoaded: function() {
+    this.data.cursor.object3DMap.mesh.renderOrder = 1;
+  },
+
+  _handleControllerConnected: function(e) {
+    const data = {
+      controller: e.target,
+      handedness: e.detail.component.data.hand
+    };
+
+    if (data.handedness === this.data.handedness) {
+      this.controllerQueue.unshift(data);
+    } else {
+      this.controllerQueue.push(data);
+    }
+
+    this._updateController();
+  },
+
+  _handleControllerDisconnected: function(e) {
+    for (let i = 0; i < this.controllerQueue.length; i++) {
+      if (e.target === this.controllerQueue[i].controller) {
+        this.controllerQueue.splice(i, 1);
+        this._updateController();
+        return;
+      }
+    }
+  },
+
+  _updateController: function() {
+    this.hasPointingDevice = this.controllerQueue.length > 0 && this.inVR;
+
+    this.el.setAttribute("line", { visible: this.hasPointingDevice });
```

Comment:
This gets set in a bunch of different places, I am betting there are ways to make it go out of sync. Should consolidate all to one spot that takes into account `hasPointingDevice` and the visibility state of the logical cursor (as in the sphere + the line + any other things that may make up what the user sees that we add later)